src/uart.rs

```diff
@@ -0,0 +1,331 @@
+            Self::Odd => uart_parity_t_UART_PARITY_ODD,
+            Self::Mark => uart_parity_t_UART_PARITY_MARK,
+            Self::Space => uart_parity_t_UART_PARITY_SPACE,
+        }
+    }
+}
+
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum StopBits {
+    One,
+    Two,
+}
+
+impl StopBits {
+    fn to_c(self) -> uart_stop_bits_t {
+        match self {
+            Self::One => uart_stop_bits_t_UART_STOP_BITS_1,
+            Self::Two => uart_stop_bits_t_UART_STOP_BITS_2,
+        }
+    }
+}
+
+/// This struct contains the `UART` device and handles all operation regarding it
+///
+/// [UART implementation]: https://doc.riot-os.org/group__drivers__periph__uart.html
+#[derive(Debug)]
+pub struct UartDevice {
+    dev: uart_t,
+}
+
+impl UartDevice {
+    /// Tries to initialize the given `UART`. Returns a Result with rather `Ok<Self>` if the UART was initialized successfully or a
+    /// `Err<UartDeviceStatus>` containing the error
+    ///
+    /// # Arguments
+    ///
+    /// * `dev` - The uart_t handle to the hardware device
+    /// * `baud` - The used baud rate
+    /// * `user_callback` The user defined callback that gets called from the os whenever new data is received from the `UART`
+    ///
+    /// # Examples
+    /// ```
+    /// use riot_wrappers::uart::UartDevice;
+    /// let mut cb = |new_data| {
+    ///     //do something here with the received data
+    /// };
+    /// let mut uart = UartDevice::new(uart_type_t_STM32_USART, 115200, &mut cb)
+    ///     .unwrap_or_else(|e| panic!("Error initializing UART: {e:?}"));
+    /// uart.write(b"Hello from UART");
+    /// ```
+    pub fn new<F>(
+        dev: uart_t,
+        baud: u32,
+        user_callback: &'static mut F,
+    ) -> Result<Self, UartDeviceError>
+    where
+        F: FnMut(u8) + Sync + 'static,
+    {
+        unsafe {
+            uart_init(
+                dev,
+                baud,
+                Some(Self::new_data_callback::<F>),
+                user_callback as *mut _ as *mut c_void,
+            )
+        }
+        .negative_to_error()
+        .map(|_| Self { dev })
+        .map_err(UartDeviceError::from_c)
+    }
+
+
+    /// Tries to initialize the given `UART`. Returns a Result with rather `Ok<Self>` if the UART was initialized successfully or a
+    /// `Err<UartDeviceStatus>` containing the error. As the name implies, the created `UART` device can <b>ONLY</b> send data
+    ///
+    /// # Arguments
+    ///
+    /// * `dev` - The uart_t handle to the hardware device
+    /// * `baud` - The used baud rate
+    ///
+    /// # Examples
+    /// ```
+    /// use riot_wrappers::uart::UartDevice;
+    /// let mut uart = UartDevice::new_without_rx(uart_type_t_STM32_USART, 115200)
+    /// .unwrap_or_else(|e| panic!("Error initializing UART: {e:?}"));
+    /// uart.write(b"Hello from UART");
+    /// ```
+    pub fn new_without_rx(dev: uart_t, baud: u32) -> Result<Self, UartDeviceError> {
```

Comment:
Could you swap out the `uart_t` with a usize index, following the plan inhttps://github.com/RIOT-OS/rust-riot-wrappers/issues/37#issuecomment-1430112835 ?

(Same goes in the `new` function)

Reply:
Done :)